### PR TITLE
Introduce VM.assert_can_migrate_sender

### PR DESCRIPTION
**Pull request by robhoes**

### CODE REVIEW

File: ocaml/idl/datamodel.ml [hunk 1]
```diff
@@ -2476,6 +2476,21 @@ let vm_assert_can_migrate = call
     ~errs:[Api_errors.license_restriction]
     ()
 
+let vm_assert_can_migrate_sender = call
+    ~name:"assert_can_migrate_sender"
+    ~lifecycle:[]
```
Comment (mseri) [inline ocaml/idl/datamodel.ml]: We will need to remember to fill in the lifecycle here

**Comment by mseri** [comment]
Do we need to add the call also to https://github.com/xapi-project/xen-api/pull/3043/files#diff-9a639b670fb7e2432894cc91df8b27fbR1706 or it is not relevant there?

**Comment by robhoes** [comment]
Oh, you're right...

**Comment by robhoes** [comment]
@mseri Thanks, updated.

**Pull request merged**